packages/server/billing/helpers/teamLimitsCheck.ts

```diff
@@ -158,4 +175,34 @@ export const checkTeamsLimit = async (orgId: string, dataLoader: DataLoaderWorke
     await enableUsageStats(billingLeadersIds, orgId)
     await sendWebsiteNotifications(orgId, billingLeadersIds, dataLoader)
   }
+
+  await scheduleJobs(scheduledLockAt, orgId)
+}
+
+export const processLockOrganizationJob = async (
+  job: ScheduledJobOrganizationLock,
+  dataLoader: DataLoaderWorker
+) => {
+  const {orgId, runAt} = job
+
+  const organization = await dataLoader.get('organizations').load(orgId)
+
+  // Skip the job if unlocked or already locked or scheduled lock date changed
+  if (
+    !organization.scheduledLockAt ||
+    organization.lockedAt ||
+    organization.scheduledLockAt.getTime() !== runAt.getTime()
+  ) {
+    return
+  }
+
+  const now = new Date()
+
+  return r
+    .table('Organization')
+    .get(orgId)
+    .update({
+      lockedAt: now
+    })
+    .run()
```

Comment:
+1 I think the file should be broken up into separate files as it's becoming quite long and difficult to follow

Reply:
changed here #7637

Reply:
-1 You're loading the organization with the dataloader in l188, so we should invalidate or update it in the cache because we don't know what will be run after this function, there might be more org related things scheduled.